src/transformations/fold.ts

```diff
@@ -47,6 +48,42 @@ function makeNumFold<T>(
   };
 }
 
+export async function genericFold(
+  dataset: DataSet,
+  base: string,
+  expression: string,
+  inputColumnName: string,
```

Comment:
Do we need an `inputColumnName` at all? This made sense for other folds, since they were restricted to folding over a column. Since we're allowing the user to supply an expression, they can access the whole row, so I don't think we need them to indicate a particular column.